hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcConnection.java

```diff
@@ -303,12 +309,16 @@ protected void initChannel(Channel ch) throws Exception {
       .addListener(new ChannelFutureListener() {
 
         private void succeed(Channel ch) throws IOException {
-          ch.writeAndFlush(connectionHeaderPreamble.retainedDuplicate());
+          if (connectionRegistryCall != null) {
+            getConnectionRegistry(ch);
+            return;
+          }
+          NettyFutureUtils.safeWriteAndFlush(ch, connectionHeaderPreamble.retainedDuplicate());
           if (useSasl) {
             saslNegotiate(ch);
           } else {
             // send the connection header to server
-            ch.write(connectionHeaderWithLength.retainedDuplicate());
+            NettyFutureUtils.safeWrite(ch, connectionHeaderWithLength.retainedDuplicate());
```

Comment:
just checking -- was this change just a cleanup you noticed, or related to the change?

Reply:
Just some error prone warning on 'future return value ignored', not critical.